viewing_party/party.py

```diff
@@ -1,23 +1,169 @@
+            max_count = num
+            max_movie_genre = genre
+    return max_movie_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+
+    movie_friends_watched = []
+    unique_movie = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            movie_friends_watched.append(movie["title"])
+
+    for movie in user_data["watched"]:
+        if movie["title"] not in movie_friends_watched:
+            unique_movie.append(movie)
+    return unique_movie
+
+def get_friends_unique_watched(user_data):
+
+    movie_friends_watched = []
+    user_not_watched = []
+    user_watched = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            movie_friends_watched.append(movie)
+
+    for movie in user_data["watched"]:
+        user_watched.append(movie["title"])
+
+    for movie in movie_friends_watched:
+        if movie["title"] not in user_watched and movie not in user_not_watched:
+            user_not_watched.append(movie)
+    return user_not_watched
+  
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    
+    recomme_movie_list = []
+    user_watched_set = set()
+
+    for movie in user_data["watched"]:
+        user_watched_set.add(movie["title"])
+    
+    unique_title_set = set()
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in user_watched_set:
+                    if movie["host"] in user_data["subscriptions"]:
+                        if movie["title"] not in unique_title_set:
+                            recomme_movie_list.append(movie)
+                            unique_title_set.add(movie["title"])
+
+    return recomme_movie_list
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+
+    if not user_data["watched"]: 
+        return []
+    
+    most_freq_genre = get_most_watched_genre(user_data)
+    recomme_movie_full_list = get_available_recs(user_data)
+
+    recomme_movie_list = []
+
+    for movie in recomme_movie_full_list:
+        if movie["genre"] == most_freq_genre:
+            recomme_movie_list.append(movie)
+            
+    return recomme_movie_list
+
+
+def get_rec_from_favorites(user_data):
+
+    recommended_movies = []
+    friends_watched = set()
+    
+    for friend in user_data.get("friends", []):
+        for movie in friend.get("watched", []):
+            friends_watched.add(movie["title"])
+    
+    for movie in user_data.get("favorites", []):
+        if movie["title"] not in friends_watched:
+            recommended_movies.append(movie)

```

Comment:
LIke with `get_available_recs` you are have a function that looks gives you the list of movies that user has watched but not their friends. This is another opportunity to make your code DRY. @daniellew712 @Allisonalex @omatesh